recipes-devtools/hackage/hkg-network/hkg-network_3.1.2.7.bb

```diff
@@ -7,4 +7,9 @@ inherit hackage
 SRC_URI[md5sum] = "f32e287c9957b6605da3dd677182cb68"
 SRC_URI[sha256sum] = "7f7620fef1a1af3d3d6747f510e73223a5c600e7d7fd9ace073d1222bdc63d85"
 
+FILESEXTRAPATHS_prepend := "${THISDIR}/patches:"
+SRC_URI += "\
+    file://expose-needed-modules.patch \
+"
+
```

Comment:
Maybe this should be dropped, and xclibs/xchargo should just be hardcoded to create a stream socket:
https://github.com/OpenXT/xclibs/blob/66d6b98010fca05a4ed6d1850f61dbd79914646b/xchargo/Tools/Argo.hsc#L90

I think there are only a few instances:
https://github.com/OpenXT/xclibs/blob/66d6b98010fca05a4ed6d1850f61dbd79914646b/xch-rpc/Rpc/DBusArgo.hs#L61
and two in manager.git xec and rpc-proxy.

Reply:
Something like:
```
socket :: SocketType -> IO Fd
socket Stream = -- ~existing with (packSocketType t) hardcoded as 1
socket _ = error
```

/usr/include/bits/socket_type.h:  SOCK_STREAM = 1,		/* Sequenced, reliable, connection-based

Reply:
I'd have to get it building and try it out.  It would probably work, just can't say if it would  take awhile to wrestle with the type system.

Reply:
I tested this with ghc6, and it seems to work: https://github.com/jandryuk/xclibs/commit/2285206b88060ec10d280cfb349adc2cc790e289

Reply:
Ok I have something that builds.  Deployed it and the build seems fine.  The other instances you described are importing the SocketType properly to be able to use "Stream" directly, so no changes needed there.